### ISSUE TITLE
Problem with uploadFile on openVMS server

### ISSUE THREAD

**Issue by BenJack57140** [robinrodricks/FluentFTP#424]
**FTP OS:

**FTP Server: OpenVMS

**Computer OS: windows 10

<if the remotePath is "/"+File name, this file is upload but the name is prefixed with an "_"
 if the remotePath is ""+ file name or anything + file name  the method returns syntax error (see logs)> 

**Logs :**

```
# Connect()
Status:   Connecting to ***:21
Response: 220 ******* FTP Server (Version 5.3) Ready.
Command:  USER ***
Response: 331 Username ******requires a Password
Command:  PASS ***
Response: 230 User logged in.
Command:  FEAT
Response: 502 FEAT is unimplemented.
Status:   Text encoding: System.Text.ASCIIEncoding
Command:  SYST
Response: 200 VMS OpenVMS V7.3 on node ********.
Status:   Detected FTP server: OpenVMS
Status:   Auto-detected VMS listing parser

# UploadFile("E:\Release\exportFTP\73000023.RES", "\73000023.RES", Overwrite, False, None)

# FileExists("\73000023.RES")

# GetNameListing("/")
Command:  TYPE I
Response: 200 TYPE set to IMAGE.

# OpenPassiveDataStream(AutoPassive, "NLST /", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||63414|)
Status:   Connecting to ***:63414
Command:  NLST /
Response: 150 Opening data connection for / (139.53.211.148,65047)
Status:   Disposing FtpSocketStream...

# CloseDataStream()
Response: 226 NLST Directory transfer complete
Status:   Disposing FtpSocketStream...

# OpenWrite("\73000023.RES", ASCII)
Command:  TYPE A
Response: 200 TYPE set to ASCII.

# OpenPassiveDataStream(AutoPassive, "STOR /73000023.RES", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||63415|)
Status:   Connecting to ***:63415
Command:  STOR /73000023.RES
Response: 150 Opening data connection for FTP_DEFAULT:[WAGN_IN]_73000023.RES; (139.53.211.148,65048)
Status:   Disposing FtpSocketStream...
Response: 226 Transfer complete.

# Dispose()
Status:   Disposing FtpClient object...
Command:  QUIT
Response: 221 Goodbye.
Status:   Disposing FtpSocketStream...
Status:   Disposing FtpSocketStream...


# Connect()
Status:   Connecting to ***:21
Response: 220 ******* FTP Server (Version 5.3) Ready.
Command:  USER ***
Response: 331 Username ***** requires a Password
Command:  PASS ***
Response: 230 User logged in.
Command:  FEAT
Response: 502 FEAT is unimplemented.
Status:   Text encoding: System.Text.ASCIIEncoding
Command:  SYST
Response: 200 VMS OpenVMS V7.3 on node *******.
Status:   Detected FTP server: OpenVMS
Status:   Auto-detected VMS listing parser

# UploadFile("E:\Release\exportFTP\73000025.RES", "73000025.RES", Overwrite, False, None)

# FileExists("73000025.RES")

# GetWorkingDirectory()
Command:  PWD
Response: 257 "FTP_DEFAULT:[WAGN_IN]" is current directory.

# GetNameListing("FTP_DEFAULT:[WAGN_IN]")

# GetWorkingDirectory()
Command:  PWD
Response: 257 "FTP_DEFAULT:[WAGN_IN]" is current directory.
Command:  TYPE I
Response: 200 TYPE set to IMAGE.

# OpenPassiveDataStream(AutoPassive, "NLST FTP_DEFAULT:[WAGN_IN]/FTP_DEFAULT:[WAGN_IN]", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||63435|)
Status:   Connecting to ***:63435
Command:  NLST FTP_DEFAULT:[WAGN_IN]/FTP_DEFAULT:[WAGN_IN]
Response: 550 file specification syntax error
Status:   Disposing FtpSocketStream...

# CloseDataStream()

# Dispose()
Status:   Disposing FtpClient object...
Command:  QUIT
Response: 221 Goodbye.
Status:   Disposing FtpSocketStream...
Command:  STOR /73000023.RES
Response: 150 Opening data connection for FTP_DEFAULT:[WAGN_IN]_73000023.RES; (139.53.211.148,65048)
Status:   Disposing FtpSocketStream...
Response: 226 Transfer complete.

# Dispose()
Status:   Disposing FtpClient object...
Command:  QUIT
Response: 221 Goodbye.
Status:   Disposing FtpSocketStream...


# Connect()
Status:   Connecting to ***:21
Response: 220 ******* FTP Server (Version 5.3) Ready.
Command:  USER ***
Response: 331 Username ***** requires a Password
Command:  PASS ***
Response: 230 User logged in.
Command:  FEAT
Response: 502 FEAT is unimplemented.
Status:   Text encoding: System.Text.ASCIIEncoding
Command:  SYST
Response: 200 VMS OpenVMS V7.3 on node ********.
Status:   Detected FTP server: OpenVMS
Status:   Auto-detected VMS listing parser

# UploadFile("E:\Release\exportFTP\73000026.RES", "./73000026.RES", Overwrite, False, None)

# FileExists("./73000026.RES")

# GetWorkingDirectory()
Command:  PWD
Response: 257 "FTP_DEFAULT:[WAGN_IN]" is current directory.

# GetNameListing("FTP_DEFAULT:[WAGN_IN]")

# GetWorkingDirectory()
Command:  PWD
Response: 257 "FTP_DEFAULT:[WAGN_IN]" is current directory.
Command:  TYPE I
Response: 200 TYPE set to IMAGE.

# OpenPassiveDataStream(AutoPassive, "NLST FTP_DEFAULT:[WAGN_IN]/FTP_DEFAULT:[WAGN_IN]", 0)
Command:  EPSV
Response: 229 Entering Extended Passive Mode (|||56097|)
Status:   Connecting to ***:56097
Command:  NLST FTP_DEFAULT:[WAGN_IN]/FTP_DEFAULT:[WAGN_IN]
Response: 550 file specification syntax error
Status:   Disposing FtpSocketStream...

# CloseDataStream()

# Dispose()
Status:   Disposing FtpClient object...
Command:  QUIT
Response: 221 Goodbye.>
```

**Comment by robinrodricks** [comment]
3rd time I'm fixing this. Used a proper Regex implementation now. Hope it solves this.

Released in https://www.nuget.org/packages/FluentFTP/25.0.6

**Comment by BenJack57140** [comment]
Hello,

Big thanks for the fix. It’s working now.

De : Robinicks [mailto:notifications@github.com]
Envoyé : jeudi 11 juillet 2019 13:52
À : robinrodricks/FluentFTP
Cc : JACQUES Benoit; Author
Objet : Re: [robinrodricks/FluentFTP] Problem with uploadFile on openVMS server (#424)


3rd time I'm fixing this. Used a proper Regex implementation now. Hope it solves this.

Released in https://www.nuget.org/packages/FluentFTP/25.0.6<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.nuget.org%2Fpackages%2FFluentFTP%2F25.0.6&data=02%7C01%7Cbenoit.jacques%40preciamolen.fr%7Cf39d148003824a06d23908d705f633f5%7C9281c83e132c4ab8be8034517d7a5b70%7C0%7C0%7C636984427314208457&sdata=Y%2Bi0rFuux0e29ErL%2FKCeiE1AtRphWoKLtIaLl1Ou%2BvU%3D&reserved=0>

**Comment by robinrodricks** [comment]
Released in 26.0.0.

https://www.nuget.org/packages/FluentFTP/26.0.0
